hw/ip/sysrst_ctrl/doc/_index.md

```diff
@@ -0,0 +1,95 @@
+
+## Keyboard and input triggered interrupt
+
+Pwrb, key0, key1, key2, ac_present and ec_rst_l can trigger interrupt on rising or falling edges.
+The SW defines the buttons, keys or inputs and their edges(H->L or L->H) that can trigger the interrupt in key_intr_ctl. 
+For example, when the power button is pressed, pwrb goes from logic “1” to logic “0” and it is a H->L transition. When the power button is released, pwrb goes from logic “0” to logic “1” and it is a L->H transition.
+The SW defines the debounce timer in key_intr_debounce_ctl
+When sysrst_ctrl detects a transition(H->L or L->H) as specified in key_intr_ctl and it meets the debounce requirement in key_intr_debounce_ctl, sysrst_ctrl sets the intr_status to indicate the source.
+It’s possible for SW to program both H->L and L->H transition detection so that it can monitor that a button is pressed first and then released
+sysrst_ctrl will drive the consolidated interrupt to PLIC
+The interrupt handler will read intr_status and write “logic 1” to clear the interrupt
+
+## Pin input value accessibility
+
+sysrst_ctrl allows the SW to read the raw input pin value(before the invert logic). The hardware writes into pin_in_value for pwrb_in, key0/1/2_in, ac_present, ec_rst_l. The software can read them like GPIO.
+
+## Pin output and keyboard inversion control
+
+sysrst_ctrl allows the SW to change the keyboard's polarity on the inputs and outputs. It also provides the output pin override feature. In other words, SW can override the hardware state machine. The assertion has higher priority. Which output pins to override and the value to drive are configurable and lockable.
+The SW defines the allowed output pin override value in pin_allowed_ctl. For example, ec_rst_l_0=1’b1 and ec_rst_l_1=1’b0 means SW allows ec_rst_l to be overridden with logic 0, not logic 1.
+If the SW locks the configuration with cfg_wen_ctl, pin_allowed_ctl cannot be modified until GSC is reset
+When the system is up and running, the SW can modify pin_out_ctl and pin_out_value to enable or disable the feature. For example, to release ec_rst_l after GSC completes the reset, SW will program pin_out_ctl[1]=1’b0. The hardware will stop driving ec_rst_l=1’b0.
+The SW defines the keyboard polarity in key_invert_ctl. For inputs, it will be inverted before the combo logic. For outputs, it will be inverted after the pinout override logic.
+
+## EC and Power-on-reset
+
+GSC and EC will be reset together during power-on. When GSC is in reset, ec_rst_l will be asserted(active low).
+When GSC exits from reset, ec_rst_l should still be asserted. The power-on-reset value of pin_allowed_ctl[1] and pin_out_ctl[1] will guarantee ec_rst_l=1’b0 after GSC reset.
+The SW will release ec_rst_l explicitly by writing pin_out_ctl[1]=1’b0 after we enter the GSC-RW boot stage
+Now both GSC and EC complete the power-on-reset
```

Comment:
Missing periods at the end of the sentences.

Reply:
Sure.